### PR TITLE
Use packagereference on cache tests and minor test fix

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: tests/CacheCompat/CommonCache.Test.AdalV4/CommonCache.Test.AdalV4.csproj [hunk 1]
```diff
@@ -1,80 +1,32 @@
...
+    <OutputType>Library</OutputType>
+    <TargetFramework>net461</TargetFramework>
+
   </PropertyGroup>
+
   <ItemGroup>
-    <PackageReference Include="CommandLineParser">
-      <Version>2.4.3</Version>
-    </PackageReference>
     <PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory">
```
Comment (MarkZuber) [inline tests/CacheCompat/CommonCache.Test.AdalV4/CommonCache.Test.AdalV4.csproj]: Should be able to make Version="4.5.1" inline as an attribute here too.

File: tests/devapps/NetCoreTestApp/NetCoreTestApp.csproj [hunk 2]
```diff
@@ -1,13 +1,15 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <OutputType>Exe</OutputType>
+    <OutputType>Library</OutputType>
```
Comment (MarkZuber) [inline tests/devapps/NetCoreTestApp/NetCoreTestApp.csproj]: Why is this now a library?  It's an application.

File: tests/CacheCompat/CommonCache.Test.AdalV5/CommonCache.Test.AdalV5.csproj [hunk 3]
```diff
@@ -1,80 +1,34 @@
...
-      <Name>CommonCache.Test.Common</Name>
-    </ProjectReference>
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
```
Comment (MarkZuber) [inline tests/CacheCompat/CommonCache.Test.AdalV5/CommonCache.Test.AdalV5.csproj]: Do we still need the LangVersion property in here and others?  it was 7.1 below...

**Comment by henrik-me** [comment]
Would love a reference or more info on how this all works.

**Pull request merged**